### DISCUSSION TITLE
Request-PnPReIndexWeb: You are not allowed to rename this library.

### DISCUSSION THREAD

**Discussion by Ofer-Gal** [pnp/powershell#2771] [Q&A]
In script I change metadata values for files in a document library then reindex using 
`Request-PnPReIndexWeb -Connection $projectConn`
the connection is using app registration with certificate.

When I run in DEV tenant all is OK
When I run in Production I get an error (after a second or 2) 
 > Request-PnPReIndexWeb: You are not allowed to rename this library

I am not renaming any library. 
What can be the error here?
Thanks

**Comment by AndersRask** [comment]
Do you have scripting enabled on DEV and not PROD (the lovely double negation property named `DenyAddAndCustomizePages 0 `) on tenant or site level?
https://learn.microsoft.com/en-us/sharepoint/allow-or-prevent-custom-script 

Another issue could be, that you do not have permissions on all lists in web on PROD to set a propertybag value.

Reindexing is really just a propertybag that set and increment vti_searchversion on the list, to make it "dirty" to get picked up by changelog for crawler.

hth
Anders

**Reply by Ofer-Gal** [reply]
No custom script enabled on Dev or Prod.
I did see a difference in the Api permissions of the app registration that is running the script.
Would you be able to tell me which permission is missing? Is it SharePoint or Graph?

Thank you in advance.

**Comment by AndersRask** [comment]
I can see ExecuteQueryAsync() so assume SharePoint CSOM

You could try and see if you are able to set a propertybag value as a perm simple test 

`man Set-PnPPropertyBagValue -ex`

**Reply by Ofer-Gal** [reply]
For: 
Set-PnPPropertyBagValue -Connection $projectConn -Key "OferTest" -Value (Get-Date -Format "yyyy-MM-ddThh:mm:ssZ")
I get:
Set-PnPPropertyBagValue: Access is denied. (Exception from HRESULT: 0x80070005 (E_ACCESSDENIED)) Site might have NoScript enabled, this prevents setting some property bag values.

Nothing about rename this library 😒

**Reply by AndersRask** [reply]
Alas, you cannot always (seldom!) trust the error messages that bubbles up in SharePoint API ;-)

Could you double-check that the *site* in DEV doesn't have scripting enabled? 
Your example kinda proves that this is a requirement for setting re-index.
Remember that script can be enabled on both tenant and site level

**Reply by Ofer-Gal** [reply]
I get the same in dev as in Production:

> Set-PnPPropertyBagValue: Access is denied. (Exception from HRESULT: 0x80070005 (E_ACCESSDENIED)) Site might have NoScript enabled, this prevents setting some property bag values.

Must be API permission

**Reply by AndersRask** [reply]
```
PS > set-pnpsite -NoScriptSite $true
PS > Request-PnPReIndexWeb
Request-PnPReIndexWeb: Access is denied. (Exception from HRESULT: 0x80070005 (E_ACCESSDENIED))
PS > set-pnpsite -NoScriptSite $false
PS > Request-PnPReIndexWeb
PS >
```
Try and enable script and it should work

**Reply by Ofer-Gal** [reply]
Tried and got the same error 😊

**Comment by automationx-work** [comment]
Have any one got answer we have 1000s of site required reindex and keep throwing error.

Request-PnPReIndexWeb : You are not allowed to rename this library.
At line:3 char:5
+     Request-PnPReIndexWeb -Web $web
+     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : InvalidOperation: (:) [Request-PnPReIndexWeb], PSInvalidOperationException
    + FullyQualifiedErrorId : InvalidOperation,PnP.PowerShell.Commands.RequestReIndexWeb

**Reply by thechiefn** [reply]
I'm getting the same error message.  The site I question we are working on was a tenant-to-tenant migrated site.  The reindex seems to never actually occur.
